v1/ratelimit/redis_delayed_sync_pipelined_test.go

```diff
@@ -460,3 +461,52 @@ func Test_GetResetAt_DefaultIsZero_Pipelined(t *testing.T) {
 		t.Fatalf("expected resetAt to be 0 for unused key, got %d", got)
 	}
 }
+
+// Note: this test will most of the time pass due to the data race it tests against being very hard to replicate
```

Comment:
When I test for flaky behaviour I have a loop with thousands of iterations inside the test and expect each and every one to succeed. Will this approach work here?